src/roswire/ros2/node_manager.py

```diff
@@ -17,7 +17,7 @@
 @attr.s(frozen=True, auto_attribs=True, slots=True)
 class ROS2NodeManager(Mapping[str, ROS2Node]):
     """Provides an interface for interacting with ROS2 nodes."""
-    app_instance: 'AppInstance'
+    app_instance: 'AppInstance' = attr.ib()
```

Comment:
Just a heads up: `attr.ib()` isn't actually needed here since we use `auto_attribs=True` at the top. Using either approach is fine, but if we're going to use `attr.ib()`, we probably want to remove the `auto_attribs=True` keyword to some unexpected accidents involving class vars.